client/app/components/packages/pas-form/edit.hbs

```diff
@@ -693,35 +693,21 @@
     <div class="cell large-4">
 
       <div class="sticky-sidebar">
-        <p>
-          <button
-            type="button"
-            class="button expanded large"
-            disabled={{not this.isSaveable}}
-            {{on "click" (perform this.save)}}
-            data-test-save-button
-          >
-            {{!-- TODO: Make this button work with the action passed in @save--}}
-            {{#if this.save.isRunning}}
-              <FaIcon @icon="spinner" @spin="true" size="5x" class="light-gray" />
-            {{else}}
-              Save
-            {{/if}}
-          </button>
-          <button
-            type="button"
-            class="button expanded large secondary"
-            disabled={{not this.isSubmittable}}
-            {{on "click" this.toggleModal}}
-            data-test-submit-button
-          >
-            {{#if this.submit.isRunning}}
-              <FaIcon @icon="spinner" @spin="true" size="5x" class="light-gray" />
-            {{else}}
-              Submit
-            {{/if}}
-          </button>
-        </p>
+        <Packages::save-button
+          @clickEvent={{perform this.save}}
+          @isEnabled={{this.isSaveable}}
+          @isRunning={{this.save.isRunning}}
+          @isSaved={{and (not this.isSaveable) (not this.save.isRunning)}}
```

Comment:
this seems like a lot of state to be passing down. I wonder how much of these conditionals can be extracted from pas-form/edit.js itself? Probably can't do it yet.

To reduce the public api surface of this component, can we just pass the `save` task? A requirement of the button is that it expects a "task" type object. They all have the same API.

When we have that inside this component, then we can derive all that state:

```handlebars
<Packages::SaveButton
  @onClick={{perform this.save}}
  @isEnabled={{this.isSaveable}}
  @task={{this.save}}
/>
```

```javascript
class SaveButton {
  get isSaved() {
    return (!this.args.isEnabled && !this.save.isRunning)
  }
}
```

Reply:
I tried it that way and it got… complicated. I discovered that `@isRunning` and `@isSaved` args made everything simpler. 

I'm passing states explicitly because there are different conditions for Save and Submit — e.g. we don't want "Submitted" to turn green when disabled and finished running, so if I don't pass an `@isSaved` arg it can't turn green. If the states are computed, it makes JS necessary (the component is template-only rn). 

I thought it better to just pass the state as args because computing the states in the component JS would need to cover multiple edge cases — e.g. the two "Submit" buttons have different conditions for `@isEnabled`, one of which can't be computed off of the `@onClick`. 

Maybe we can refactor this later. Once we're dealing with multiple packages, there could be other factors that determine the state of these buttons.